### PR TITLE
Migrate other files in Core to JUnit5

### PR DESCRIPTION
**Pull request by tomtongue**
Migrate the following test classes in iceberg-core to JUnit 5 and AssertJ style for https://github.com/apache/iceberg/issues/9085.

## Current Progress 
- [x] `TestMetricsModes.java` 
- [x] `TestMicroBatchBuilder.java`
- [x] `TestRemoveSnapshots.java`
- [x] `TestRewriteManifests.java`
- [x] `TestRowDelta.java`
- [x] `TestSequenceNumberForV2Table.java`
- [x] `TestSingleValueParser.java`
- [x] `TestSortOrder.java`
- [x] `TestSortOrderParser.java`
- [x] `TestSplitPlanning.java`
- [x] `TestV1ToV2RowDeltaDelete.java` 
- [x] `V2TableTestBase.java` 

Refactoring (related to https://github.com/apache/iceberg/pull/10014#discussion_r1534263933):
- [x] `TestSchemaAndMappingUpdate.java`
- [x] `TestSchemaUpdate`

### CODE REVIEW

**Comment by tomtongue** [comment]
@nastra Could you review this PR?

The following two classes in `org/apache/iceberg` need to be migrated for now:
* `TestWapWorkflow`
* `TestMetrics` (this class is related files in `iceberg-data`)

File: core/src/test/java/org/apache/iceberg/TestSplitPlanning.java [hunk 1]
```diff
@@ -33,40 +38,34 @@
...
 
   private static final Configuration CONF = new Configuration();
   private static final HadoopTables TABLES = new HadoopTables(CONF);
   private static final Schema SCHEMA =
       new Schema(
           optional(1, "id", Types.IntegerType.get()), optional(2, "data", Types.StringType.get()));
 
-  @Rule public TemporaryFolder temp = new TemporaryFolder();
-  private Table table = null;
+  // @Rule public TemporaryFolder temp = new TemporaryFolder();
```
Comment (nastra) [inline core/src/test/java/org/apache/iceberg/TestSplitPlanning.java:55]: can be removed

File: core/src/test/java/org/apache/iceberg/TestRemoveSnapshots.java [hunk 2]
```diff
@@ -136,43 +123,37 @@ public void testExpireOlderThanWithDelete() {
...
+                firstSnapshot
+                    .allManifests(table.io())
+                    .get(0)
+                    .path(), // manifest was rewritten for delete
+                secondSnapshot.manifestListLocation(), // snapshot expired
+                secondSnapshot
+                    .allManifests(table.io())
+                    .get(0)
+                    .path(), // manifest contained only deletes, was dropped
+                FILE_A.path().toString() // deleted
```
Comment (nastra) [inline core/src/test/java/org/apache/iceberg/TestRemoveSnapshots.java:144]: why is `toString()` needed here?

File: core/src/test/java/org/apache/iceberg/TestRemoveSnapshots.java [hunk 3]
```diff
@@ -202,39 +182,32 @@ public void testExpireOlderThanWithDeleteInMergedManifests() {
...
+        .as("Should remove expired manifest lists and deleted data file")
+        .isEqualTo(
+            Sets.newHashSet(
+                firstSnapshot.manifestListLocation(), // snapshot expired
+                firstSnapshot
+                    .allManifests(table.io())
+                    .get(0)
+                    .path(), // manifest was rewritten for delete
+                secondSnapshot.manifestListLocation(), // snapshot expired
+                FILE_A.path()));
```
Comment (nastra) [inline core/src/test/java/org/apache/iceberg/TestRemoveSnapshots.java:199]: original comment is missing here: `// deleted`

File: core/src/test/java/org/apache/iceberg/TestRemoveSnapshots.java [hunk 4]
```diff
@@ -257,32 +229,30 @@ public void testExpireOlderThanWithRollback() {
...
+    assertThat(table.snapshot(secondSnapshot.snapshotId()))
+        .as("Expire should remove the orphaned snapshot")
+        .isNull();
+
+    assertThat(deletedFiles)
+        .as("Should remove expired manifest lists and reverted appended data file")
+        .isEqualTo(
+            Sets.newHashSet(
+                secondSnapshot.manifestListLocation(), // snapshot expired
+                Iterables.getOnlyElement(secondSnapshotManifests)
```
Comment (nastra) [inline core/src/test/java/org/apache/iceberg/TestRemoveSnapshots.java:245]: can we avoid using `Iterables` here?

File: core/src/test/java/org/apache/iceberg/TestRemoveSnapshots.java [hunk 5]
```diff
@@ -1102,19 +1050,17 @@ public void testExpireWithDefaultSnapshotAge() {
...
-    Assert.assertEquals("Should keep 1 snapshot", 1, Iterables.size(table.snapshots()));
-    Assert.assertEquals(
-        "Should remove expired manifest lists",
-        Sets.newHashSet(
-            firstSnapshot.manifestListLocation(), secondSnapshot.manifestListLocation()),
-        deletedFiles);
+    assertThat(table.currentSnapshot()).isEqualTo(thirdSnapshot);
+    assertThat(table.snapshots()).hasSize(1);
+    assertThat(deletedFiles)
+        .isEqualTo(
```
Comment (nastra) [inline core/src/test/java/org/apache/iceberg/TestRemoveSnapshots.java:1056]: `.containsExactly()`

**Comment by nastra** [comment]
> @nastra Could you review this PR?
> 
> Just let you know, the following two classes in `org/apache/iceberg` will be migrated:
> 
> * `TestWapWorkflow`
> * `TestMetrics` (this class is related files in `iceberg-data`)

You mean migrated in a separate PR? Also I see a few more files in `core` that need to be migrated:

junit4tests

**Comment by tomtongue** [comment]
> > @nastra Could you review this PR?
> > Just let you know, the following two classes in `org/apache/iceberg` will be migrated:
> > 
> > * `TestWapWorkflow`
> > * `TestMetrics` (this class is related files in `iceberg-data`)
> 
> You mean migrated in a separate PR? Also I see a few more files in `core` that need to be migrated:

I will submit a PR for the two classes `TestWapWorkflow` and `TestMetrics`.  These two classes are remaining classes in `org/apache/iceberg`. 

There are also some files to be migrated in some sub-packages (like `actions`, `avro` etc.) in `org/apache/iceberg`. I will check which file in the sub-packages should be migrated to JUnit5.

File: core/src/test/java/org/apache/iceberg/TestRemoveSnapshots.java (hunk 2 above)
Comment (tomtongue) [reply core/src/test/java/org/apache/iceberg/TestRemoveSnapshots.java:144]: This is not needed. Remove `toString()` here.

File: core/src/test/java/org/apache/iceberg/TestRemoveSnapshots.java (hunk 5 above)
Comment (tomtongue) [reply core/src/test/java/org/apache/iceberg/TestRemoveSnapshots.java:1056]: The order of `firstSnapshot.manifestLocation` and `secondSnapshot.manifestLocation` depends on the hashset values, so the order is not always the same. So, this part needs to use `isEqualTo`, or `containsExactlyInAnyOrder`.

Comment (nastra) [reply core/src/test/java/org/apache/iceberg/TestRemoveSnapshots.java:1056]: let's use `containsExactlyInAnyOrder` then

Comment (tomtongue) [reply core/src/test/java/org/apache/iceberg/TestRemoveSnapshots.java:1056]: Okay, thank you!

**Review by nastra — APPROVED** [review]

**Pull request merged**